### DISCUSSION TITLE
Alternative to the Maple worksheet to generate a new model

### DISCUSSION THREAD

**Discussion by phbasler** [BSMPT/BSMPT#76] [Ideas]
At the moment we have the [maple](https://github.com/phbasler/BSMPT/blob/master/Maple/CreateModel.mw) file to generate the tensors for a new model. 
It would be nice to have an alternative, using a non-commercial tool. 
One possibility would be https://www.sagemath.org/, we should collect other possibilities here.

**Comment by lisabiermann** [comment]
I propose to use WolframScript as an interface to calculate the tensors. It's free and works on any platform. 
I've attached a [MWE](https://github.com/phbasler/BSMPT/files/8621768/r2hdm_example.zip) that calculates the `Curvature_Higgs_L2` tensors for the R2HDM. 
You can run it with `wolframscript -script r2hdm_example.m`.

**Reply by phbasler** [reply]
Have you checked how this works with the license? In https://www.wolfram.com/legal/terms/wolfram-engine.html they write 
'Open-source projects approved by Wolfram.' 

Can you check what this means?

**Reply by lisabiermann** [reply]
WolframScript provides (free) access to the Wolfram Language. If we would provide an interface (for the model generation) to BSMPT written in Wolfram Language, any user would be able to use it - independent of whether or not they have access to a Mathematica licence. If you have one, the same script just runs with `math -script r2hdm_example.m`.

About the license: We would be only providing an optional interface - like we did with Maple - but now with the advantage of accessibility for everyone.
I don't see why we should run into issues with the license of WolframEngine, as the interface script would not be limited to be run only with `wolframscript` and this piece of code would also not be included in BSMPT itself, just provided aside of it.

**Comment by phbasler** [comment]
Another possibility is using sympy
